src/scalar/timeintegrate_scalar_mod.F90

```diff
@@ -17,8 +23,560 @@ SUBROUTINE timeintegrate_scalar(itstep, ittot, timeph, dt, irk, rkscheme)
         TYPE(rk_2n_t), INTENT(in) :: rkscheme
 
         ! Local variables
-        ! None...
+        INTEGER(intk) :: ilevel, l
+        REAL(realk) :: frhs, fu, dtrk, dtrki
+        TYPE(field_t), POINTER :: t, told, dt_f
+        TYPE(field_t) :: qtt, qtu, qtv, qtw
 
-        CONTINUE
+        IF (.NOT. solve_scalar) RETURN
+        CALL start_timer(400)
+        CALL start_timer(401)
+
+        ! Local temporary storage ("scrap")
+        CALL qtt%init("QTT")
+        CALL qtu%init("QTU", istag=1)
+        CALL qtv%init("QTV", jstag=1)
+        CALL qtw%init("QTW", kstag=1)
+
+        CALL qtu%init_buffers()
+        CALL qtv%init_buffers()
+        CALL qtw%init_buffers()
+        CALL stop_timer(401)
+
+        CALL start_timer(402)
+        DO l = 1, nsca
+            ! Fetch fields
+            CALL get_field(t, scalar(l)%name)
+            CALL get_field(dt_f, "D"//TRIM(scalar(l)%name))
+            CALL get_field(told, TRIM(scalar(l)%name)//"_OLD")
+
+            ! Copy to "T_OLD"
+            told%arr = t%arr
+
+            ! TSTSCA4 zeroize qtu, qtv, qtw before use internally
+            CALL tstsca4(qtu, qtv, qtw, t, scalar(l))
+
+            ! This operation apply boundary conditions to qtu, qtv, qtw ONLY!
+            ! Does not modify t-field at all!
+            DO ilevel = minlevel, maxlevel
+                CALL parent(ilevel, qtu, qtv, qtw)
+                CALL bound_scaflux%bound(ilevel, qtu, qtv, qtw, t)
+            END DO
+
+            ! fluxbalance zeroize qtt before use internally
+            CALL fluxbalance(qtt, qtu, qtv, qtw)
+
+            ! Ghost cell "flux" boundary condition applied to qtt field
+            IF (ib%type == "GHOSTCELL") THEN
+                CALL set_scastencils("P", scalar(l), qtt=qtt)
+            END IF
+
+            ! In IRK 1, FRHS is zero, therefore we do not need to zeroize
+            ! the dt field before each step
+            CALL rkscheme%get_coeffs(frhs, fu, dtrk, dtrki, irk)
+
+            ! dT_j = A_j*dT_(j-1) + QTT
+            dt_f%arr = frhs*dt_f%arr + qtt%arr
+
+            ! T_j = T_(j-1) + B_j*dT_j
+            t%arr = t%arr + dt*fu*dt_f%arr
```

Comment:
Yes, you are right. No errors in the original code. I have written it in this way because it is closer to various papers on RK-methods and in particular the one from Williamson. There the coefficients are dimensionless and the operations are written as in the revised version here.